specs/Deployment/DeploymentAPI.md

```diff
@@ -111,7 +111,55 @@ again.
 ```C#
     if (DeploymentManager.GetStatus().Status != DeploymentStatus.Ok)
     {
-        // Initialize does a status check, and if the status is not Ok it will attempt to get
+        // Initialize does a status check, and if the status is not OK it will attempt to get
+        // the WindowsAppRuntime into a good state by deploying packages. Unlike a simple
+        // status check, Initialize can sometimes take several seconds to deploy the packages.
+        // These should be run on a separate thread so as not to hang your app while the
+        // packages deploy.
+        var initializeTask = Task.Run(() => DeploymentManager.Initialize());
+        initializeTask.Wait();
+
+        // Check the result.
+        if (initializeTask.Result.Status != DeploymentStatus.Ok)
+        {
+            // The WindowsAppRuntime is in a bad state which Initialize() did not fix.
+            // Do error reporting or gather information for submitting a bug.
+            // Gracefully exit the program or carry on without using the WindowsAppRuntime.
+        }
+    }
+```
+
+### DeploymentInitializeOptions
+
+#### ForceTargetApplicationShutDown
+
+When this API may install newer version of one or more of the packages (i.e. an update) and if there
+are any currently running process(es) associated with that package(s) (in other words, if any of the
+packages to be updated are currently in use), then this API will fail installing that package. But
+this update to the packages can be forced by using DeploymentInitializeOptions object and setting
+ForceTargetApplicationShutDown option before passing it to this API. This option when set will
```

Comment:
It's not clear who the "target" application is. This is a parameter to `DeploymentManager.Initialize`. What is the target of the initialization?

* The app that requested the initialization. (I.e., this will terminate any other instances of the app.)
* The WindowsAppRuntime itself. (I.e., this will terminate *all* WinAppSDK apps.)
* The packages within the WindowsAppRuntime that need updating. (I.e., this will terminate *some* WinAppSDK apps.)

The Windows Runtime version has a pair of properties

* AddPackageOptions.ForceAppShutdown: Shuts down the app for this package, plus any apps dependent on the package.
* AddPackageOptions.ForceTargetAppShutdown: Shuts down the app for this package, but not apps dependent on the package.

I think this flag's semantics match the Windows Runtime flag, though I'm not certain. The existing documentation is quite sparse. Even the original spec for ForceAppShutdown and ForceTargetAppShutdown was silent about what the flags meant.

Also, the AddPackageOptions are in the context of adding a specific package, but DeploymentManager.Initialize has no package parameters, so when you say "one or more of the packages" or "this API will fail installing that package", I have no idea what package / packages you're talking about.

WinAppSDK developers may not know or care that there are "packages" (in the AppX sense) at all! The only "package" the WinAppSDK developer saw was the NuGet package that they added to their package references. And that is totally not the package that this documentation is talking about, is it?

Do we really need to drag WinAppSDK developers into the complex world of package management? They just want to call the new APIs. They were told it was as easy as adding this NuGet package to their project and start calling the new APIs. Nobody told them that they will also have to get a Masters degree in package management. Can we rephrase this in terms of the WindowsAppRuntime instead of in terms of packages?

(Second soapbox: I'm also not a fan of the word "target". It's usually a weasel-word that means "you know which one I mean", and then falls apart when it's not obvious "which one".)